d2l-image.js

```diff
@@ -16,66 +32,59 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-image">
 				border-radius: var(--d2l-image-border-radius);
 				object-fit: var(--d2l-image-object-fit);
 			}
-		</style>
+		`;
+	}
 
-		<iron-ajax id="imageRequest" url="[[imageUrl]]" headers="[[_headers]]" on-iron-ajax-response="_onImageResponse" on-iron-ajax-error="_onImageResponse" handle-as="blob">
-		</iron-ajax>
+	constructor() {
+		super();
+		window.d2lfetch.use({ name: 'auth', fn: auth });
+	}
 
-		<img id="image" alt$="[[alternateText]]">
-	</template>
+	render() {
+		if (!this._imageUrl) {
+			return nothing;
+		}
 
-</dom-module>`;
+		return html`
+			<img alt="${this.altText}" src="${ifDefined(this._imageUrl)}">
+		`;
+	}
 
-document.head.appendChild($_documentContainer.content);
+	updated(changedProperties) {
+		if (changedProperties.has('imageUrl')) {
+			this._loadImage();
+		}
+	}
 
-Polymer({
-	is: 'd2l-image',
+	async _loadImage() {
+		if (!this.imageUrl) {
+			return;
+		}
 
-	properties: {
-		alternateText: {
-			type: String,
-			reflectToAttribute: true,
-		},
-		imageUrl: {
-			type: String,
-			value: null
-		},
-		token: String,
```

Comment:
What's the plan for consumers now that `token` isn't required? I'm a little worried that the token they're passing also isn't the same one as the one you're getting via `d2l-fetch-auth`. Might be safer to just leave this and use it.

Reply:
Hmm, yeah given the gotchas you've mentioned, it might be safer to leave `token` as part of the consumer interface and not drastically change the authentication.